### PR TITLE
support nulls in TextNode equals

### PR DESCRIPTION
**Pull request by pjfanning**
relates to #4378 

@cowtowncoder since this is a bug - could we consider backporting this?

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/databind/node/TextNode.java [hunk 1]
```diff
@@ -164,13 +164,17 @@ public boolean equals(Object o)
...
+            return java.util.Objects.equals(otherNode._value, _value);
         }
         return false;
     }
 
     @Override
-    public int hashCode() { return _value.hashCode(); }
+    public int hashCode() {
+        // we need a hardcoded value for null _value to ensure that hash codes are stable
+        return _value == null ? -1 : _value.hashCode();
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/node/TextNode.java:176]: I think there's `Objects.hash()` or `Objects.hashCode()` convenience method?

**Comment by cowtowncoder** [comment]
@pjfanning Yes, I think this could be backported as far as `2.15` as I was thinking of pushing `2.15.4` in near future. Seems like a kind of thing that'd be good to fix.

**Comment by pjfanning** [comment]
@cowtowncoder switched to Objects.hashCode

File: src/main/java/com/fasterxml/jackson/databind/node/TextNode.java (hunk 1 above)
Comment (pjfanning) [reply src/main/java/com/fasterxml/jackson/databind/node/TextNode.java]: done

**Review by cowtowncoder — APPROVED** [review]

**Pull request merged**